astropy/io/fits/tests/test_connect.py

```diff
@@ -329,3 +329,13 @@ def test_unit_warnings_read_write(tmpdir):
     with catch_warnings() as l:
         Table.read(filename, hdu=1)
     assert len(l) == 0
+
+
+def test_convert_comment_convention(tmpdir):
+    """
+    Regression test for https://github.com/astropy/astropy/issues/6079
+    """
+    filename = DATA + '/table_with_comment.fits'
```

Comment:
@taldcroft I did not write temp file as other tests because such a test is not 'unit', i.e.
```
Write 'comments' to file through a writer, assume it will convert to 'COMMENT'
Read file using a reader
Expect reader have 'comments' key
```
In this case, if the writer doesn't convert 'comments' to 'COMMENT', and the reader doesn't convert 'COMMENT' to 'comments', the test still not fail, because the writer writes 'comments' to file, the reader reads 'comments' back in. So nothing really assert that the reader indeed did the conversion.